website/blog/2025-04-18-replacing-mongodb-with-ferretdb-librechat.md

```diff
@@ -0,0 +1,113 @@
+
+![Replacing MongoDB with FerretDB on LibreChat](/img/blog/ferretdb-librechat.jpg)
+
+[LibreChat](https://www.librechat.ai/) is a free, open-source application that provides a user-friendly and customizable interface for interacting with various AI providers.
+It allows users to connect with providers like [OpenAI](https://openai.com/), [Azure](https://azure.microsoft.com/), [Anthropic](https://www.anthropic.com/), and more.
+
+For LibreChat users who want to stay fully open source, FerretDB is a great drop-in replacement for MongoDB.
+
+If you're looking to avoid proprietary databases or vendor lock-in, FerretDB is a drop-in replacement for MongoDB that runs on top of PostgreSQL.
+It uses PostgreSQL with DocumentDB extension as the backend, while letting you use familiar MongoDB operations and commands.
+
+This guide shows how to run LibreChat with FerretDB as the database, either by connecting to an existing FerretDB instance or running everything in Docker.
+
+<!--truncate-->
+
+## How to use FerretDB with LibreChat
+
+LibreChat can use FerretDB as its database in two ways:
+
+### Option 1: Connect to an existing FerretDB instance
+
+If you already have FerretDB running, simply swap out MongoDB connection string with your FerretDB connection string using the `MONGO_URI` environment variable.
+
+For Docker-based setups, update the `docker-compose.override.yml` file:
+
+```yaml
+services:
+  api:
+    environment:
+      - MONGO_URI=mongodb://<username>:<password>@<host>:<port>/LibreChat
+```
+
+For local development with `npm`, update the `.env` file to point to your FerretDB instance instead of MongoDB.
+
+For example:
+
+```text
+MONGO_URI=mongodb://<username>:<password>@<host>:<port>/LibreChat
+```
+
+:::note
+
+If you're new to FerretDB, you can find [installation instructions here](https://docs.ferretdb.io/installation/ferretdb/).
+
+:::
+
+### Option 2: Add FerretDB and PostgreSQL via Docker Compose
+
+If you don't have FerretDB running, you can set compile everything together using Docker Compose.
+
+To do that, add FerretDB and PostgreSQL with DocumentDB extension to your `docker-compose.override.yml` file.
+
+Here's an example:
+
+```yaml
+services:
+  api:
+    environment:
+      - MONGO_URI=mongodb://<username>:<password>@ferretdb:27017/LibreChat
+
+  postgres:
+    image: ghcr.io/ferretdb/postgres-documentdb:17-0.102.0-ferretdb-2.1.0
+    platform: linux/amd64
+    restart: on-failure
+    environment:
+      - POSTGRES_USER=<username>
+      - POSTGRES_PASSWORD=<password>
+      - POSTGRES_DB=postgres
+    volumes:
+      - ./data:/var/lib/postgresql/data
+
+  ferretdb:
+    image: ghcr.io/ferretdb/ferretdb:2.1.0
+    restart: on-failure
+    ports:
+      - 27017:27017
+    environment:
+      - FERRETDB_POSTGRESQL_URL=postgres://<username>:<password>@postgres:5432/postgres
+```
+
+Replace `<username>` and `<password>` with your desired FerretDB credentials.
+
+Once set up, run `docker-compose up` to start the entire stack.
```

Comment:
I think we should use `docker compose up` right? https://docs.docker.com/compose/releases/migrate/#docker-compose-vs-docker-compose